src/Fixer/Basic/Psr4Fixer.php

```diff
@@ -12,89 +12,57 @@
 
 namespace PhpCsFixer\Fixer\Basic;
 
-use PhpCsFixer\AbstractPsrAutoloadingFixer;
-use PhpCsFixer\FixerDefinition\FileSpecificCodeSample;
+use PhpCsFixer\AbstractProxyFixer;
+use PhpCsFixer\Fixer\DeprecatedFixerInterface;
 use PhpCsFixer\FixerDefinition\FixerDefinition;
-use PhpCsFixer\Tokenizer\Token;
-use PhpCsFixer\Tokenizer\Tokens;
 
 /**
  * @author Jordi Boggiano <j.boggiano@seld.be>
  * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
  * @author Bram Gotink <bram@gotink.me>
  * @author Graham Campbell <graham@alt-three.com>
+ *
+ * @deprecated
  */
-final class Psr4Fixer extends AbstractPsrAutoloadingFixer
+final class Psr4Fixer extends AbstractProxyFixer implements DeprecatedFixerInterface
 {
+    /**
+     * @var PsrAutoloadingFixer
+     */
+    private $fixer;
+
+    public function __construct()
+    {
+        $this->fixer = new PsrAutoloadingFixer();
+        parent::__construct();
+    }
+
     /**
      * {@inheritdoc}
      */
     public function getDefinition()
     {
         return new FixerDefinition(
             'Class names should match the file name.',
-            [
-                new FileSpecificCodeSample(
-                    '<?php
-namespace PhpCsFixer\FIXER\Basic;
-class InvalidName {}
-',
-                    new \SplFileInfo(__FILE__)
-                ),
-            ],
-            null,
-            'This fixer may change your class name, which will break the code that is depended on old name.'
+            \array_slice($this->fixer->getDefinition()->getCodeSamples(), 0, 1),
+            $this->fixer->getDefinition()->getDescription(),
+            $this->fixer->getDefinition()->getRiskyDescription()
```

Comment:
Not that it is going to save a ton of time, but what do you think about introducing a temporary variable for the proxied fixer's definition?